cpp/tests/copying/gather_tests.cu

```diff
@@ -348,3 +349,70 @@ TEST_F(GatherTestStr, StringColumn) {
 
     cudf::test::expect_tables_equal(expected, got->view());
 }
+
+TEST_F(GatherTestStr, Gather)
+{
+    std::vector<const char*> h_strings{ "eee", "bb", "", "aa", "bbb", "ééé" };
+    cudf::test::strings_column_wrapper strings( h_strings.begin(), h_strings.end() );
+    cudf::table_view source_table ({strings});
+
+    std::vector<int32_t> h_map{ 4,1,5,2,7 };
+    cudf::test::fixed_width_column_wrapper<int32_t> gather_map{h_map.begin(), h_map.end()};
+    auto results = cudf::experimental::detail::gather(
+           source_table,
+           gather_map,
+           false, true
+           );
+
+    std::vector<const char*> h_expected;
+    std::vector<int32_t> expected_validity;
+    for( auto itr = h_map.begin(); itr != h_map.end(); ++itr )
+    {
+        auto index = *itr;
+        if( (0 <= index) && (index < static_cast<decltype(index)>(h_strings.size())) ) {
+            h_expected.push_back( h_strings[index] );
+            expected_validity.push_back(1);
+        }
+        else {
+            h_expected.push_back( "" );
+            expected_validity.push_back(0);
+        }
+    }
+    cudf::test::strings_column_wrapper expected( h_expected.begin(), h_expected.end(),
+        expected_validity.begin());
+    cudf::test::expect_columns_equal(results->view().column(0),expected);
+}
+
+TEST_F(GatherTestStr, GatherIgnoreOutOfBounds)
+{
+    std::vector<const char*> h_strings{ "eee", "bb", "", "aa", "bbb", "ééé" };
+    cudf::test::strings_column_wrapper strings( h_strings.begin(), h_strings.end() );
+    cudf::table_view source_table ({strings});
+
+    std::vector<int32_t> h_map{ 3,4,0,0 };
+    cudf::test::fixed_width_column_wrapper<int32_t> gather_map{h_map.begin(), h_map.end()};
+    auto results = cudf::experimental::detail::gather(
+           source_table,
+           gather_map,
+           false, true
+           );
+
+    std::vector<const char*> h_expected;
+    std::vector<int32_t> expected_validity;
+    for( auto itr = h_map.begin(); itr != h_map.end(); ++itr ) {
+        h_expected.push_back( h_strings[*itr] );
+        expected_validity.push_back(1);
+    }
+    cudf::test::strings_column_wrapper expected( h_expected.begin(), h_expected.end(),
+        expected_validity.begin());
+    cudf::test::expect_columns_equal(results->view().column(0),expected);
+}
+
+TEST_F(GatherTestStr, GatherZeroSizeStringsColumn)
+{
+    cudf::column_view zero_size_strings_column( cudf::data_type{cudf::STRING}, 0, nullptr, nullptr, 0);
+    auto strings_view = cudf::strings_column_view(zero_size_strings_column);
+    thrust::device_vector<int32_t> gather_map;
+    auto results = cudf::strings::detail::gather<true>(strings_view, gather_map.begin(), gather_map.end() );
```

Comment:
I was expecting that the tests would call just the `gather` API and not the `strings::detail::gather` internal function. 
Is there a way to accomplish this without calling this internal function?